drf_recaptcha/validators.py

```diff
@@ -32,55 +54,68 @@ def testing_validation(self):
                 self.messages["captcha_invalid"], code="captcha_invalid"
             )
 
-    def set_client_ip(self, serializer_field):
+    def _get_secret_key_from_context_or_default(self, serializer_field=None) -> str:
+        if not serializer_field:
+            return self.default_recaptcha_secret_key
+
+        return serializer_field.context.get("recaptcha_secret_key", self.default_recaptcha_secret_key)
+
+    def _get_client_ip_from_context_or_default(self, serializer_field=None) -> str:
+        if serializer_field and not self.default_recaptcha_client_ip:
+            return self._get_client_ip_from_context(serializer_field)
+
+        return self.default_recaptcha_client_ip
+
+    @staticmethod
+    def _get_client_ip_from_context(serializer_field):
         request = serializer_field.context.get("request")
         if not request:
             raise ImproperlyConfigured(
                 "Couldn't get client ip address. Check your serializer gets context with request."
             )
 
-        self.recaptcha_client_ip, _ = get_client_ip(request)
+        recaptcha_client_ip, _ = get_client_ip(request)
+        return recaptcha_client_ip
 
-    def get_response(self, value: str) -> client.RecaptchaResponse:
+    def get_response(self, value: str) -> RecaptchaResponse:
```

Comment:
unused code

Reply:
It's used in [`tests/test_validator.py`](https://github.com/llybin/drf-recaptcha/blob/7ad060dca82fa470512f94ba48cf3b39a9bc657d/tests/test_validator.py#L31).

Yes, I can remove it from tests, but i was thinking some projects could call get_response() direcly, so i dont want to make breaking changes.

Reply:
Here is strange for me, why this function always uses only default_recaptcha values?

Reply:
@llybin , because it is the same functionality current `get_respone` has. I just dont want to make any breaking changes.

If you think that this function could be easily removed without any bad consequences, then okey, i'll do it.

Reply:
I'm not sure, I will check. But I see some code always uses the default key even if you send other in context

Reply:
@llybin , I think it's okey to remove this function

It's supposed to be internal.. So I hope deleting this would not break many projects.

Reply:
I'll try to look at the weekend, we can change the major version in that case

Reply:
@llybin , is this the one problem that stops this PR to be merged?

if so, may be we could remove get_response and bump major version?

Reply:
When I need some changes, I forked repo and use mine and waiting until my PR will be merged and if will be.
Sorry, I'm here, wait please. Yes all other look ok, I trying understand only that moment again.

Reply:
yep, let's remove `get_response`